specs/beacon-chain.md

```diff
@@ -348,25 +348,25 @@ def split(lst, N):
     return [lst[len(lst)*i//N: len(lst)*(i+1)//N] for i in range(N)]
 ```
 
+A helper method for readability:
+
+```python
+def clamp(minval, x, maxval):
+    return minval if x < minval else maxval if x > maxval else x
+```
+
 Now, our combined helper method:
 
 ```python
 def get_new_shuffling(seed, validators, crosslinking_start_shard):
     active_validators = get_active_validator_indices(validators)
-    if len(active_validators) >= CYCLE_LENGTH * MIN_COMMITTEE_SIZE:
-        committees_per_slot = min(len(active_validators) // CYCLE_LENGTH // (MIN_COMMITTEE_SIZE * 2) + 1, SHARD_COUNT // CYCLE_LENGTH)
-        slots_per_committee = 1
-    else:
-        committees_per_slot = 1
-        slots_per_committee = 1
-        while len(active_validators) * slots_per_committee < CYCLE_LENGTH * MIN_COMMITTEE_SIZE \
-                and slots_per_committee < CYCLE_LENGTH:
-            slots_per_committee *= 2
+    committees_per_slot = clamp(1,
```

Comment:
No need for the `clamp` here. The min value of `len(active_validators) // CYCLE_LENGTH // (MIN_COMMITTEE_SIZE * 2) + 1` is `1` due to the `+ 1` term.

Reply:
@djrtwo do you mean only using `min()`?
```python
committees_per_slot = min(
    len(active_validators) // CYCLE_LENGTH // (MIN_COMMITTEE_SIZE * 2) + 1,
    SHARD_COUNT // CYCLE_LENGTH
)
```
 There is a chance that it will return 0 if we set `SHARD_COUNT < CYCLE_LENGTH`. (e.g., initial test net?)

Reply:
You're right.

I think we should not allow `SHARD_COUNT < CYCLE_LENGTH`. It would result in more than one committee (from more than one slot) working on crosslinking a single shard. This is a similar corner case that would increase complexity of the code but for no tangible reason. Especially because these are constants that we define once for the protocol rather than ones that are in flux (like num validators)